config/spotbugs/exclude.xml

```diff
@@ -217,6 +217,12 @@
         <Method name="~.*validateAnnotations.*"/>
         <Bug pattern="UC_USELESS_OBJECT"/>
     </Match>
+    <Match>
+        <!-- MongoDB status: "False Positive", SpotBugs rank: 17 -->
+        <Class name="org.bson.codecs.kotlinx.KotlinSerializerCodec$Companion"/>
+        <Method name="~.*validateAnnotations.*"/>
+        <Bug pattern="BC_BAD_CAST_TO_ABSTRACT_COLLECTION"/>
```

Comment:
I've noticed that the CodecRegistry, while being passed to the get method of a CodecProvider, is not actually utilized in the current implementation. You can see this at [KotlinSerializerCodecProvider.kt Line 35](https://github.com/vbabanin/mongo-java-driver/blob/baebf9074c69300f7bd70abcd93802d2eaa0766f/bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/KotlinSerializerCodecProvider.kt#L35).

It raises a question: Should we consider leveraging this CodecRegistry to obtain value codecs instead of relying on hardcoded implementations?

Reply:
This still needded?